### PR TITLE
Reduce boost dependency in PhysicsTools/PatAlgos

### PR DESCRIPTION
**Pull request by camolezi**
#### PR description:
Changed boost::ptr_vector for std::vector<std::unique_ptr<>>
The code should have the same behavior.

#### PR validation:
Passed on basic runTheMatrix test.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

@vgvassilev @davidlange6

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by santocch** [comment]
+1

File: PhysicsTools/PatAlgos/src/MultiIsolator.cc [hunk 1]
```diff
@@ -77,21 +77,21 @@ void MultiIsolator::addIsolator(
 }
 
 void MultiIsolator::beginEvent(const edm::Event &event, const edm::EventSetup &eventSetup) {
-  for (boost::ptr_vector<BaseIsolator>::iterator it = isolators_.begin(), ed = isolators_.end(); it != ed; ++it) {
-    it->beginEvent(event, eventSetup);
+  for (auto it = isolators_.begin(), ed = isolators_.end(); it != ed; ++it) {
+    (*it)->beginEvent(event, eventSetup);
```
Comment (slava77) [inline PhysicsTools/PatAlgos/src/MultiIsolator.cc:81]: a for range loop could be used here

File: PhysicsTools/PatAlgos/src/MultiIsolator.cc [hunk 2]
```diff
@@ -77,21 +77,21 @@ void MultiIsolator::addIsolator(
...
   }
 }
 
 void MultiIsolator::endEvent() {
-  for (boost::ptr_vector<BaseIsolator>::iterator it = isolators_.begin(), ed = isolators_.end(); it != ed; ++it) {
-    it->endEvent();
+  for (auto it = isolators_.begin(), ed = isolators_.end(); it != ed; ++it) {
```
Comment (slava77) [inline PhysicsTools/PatAlgos/src/MultiIsolator.cc:86]: a for range loop could be used here

File: PhysicsTools/PatAlgos/src/MultiIsolator.cc [hunk 3]
```diff
@@ -77,21 +77,21 @@ void MultiIsolator::addIsolator(
...
+    (*it)->endEvent();
   }
 }
 
 void MultiIsolator::print(std::ostream &out) const {
-  for (boost::ptr_vector<BaseIsolator>::const_iterator it = isolators_.begin(), ed = isolators_.end(); it != ed; ++it) {
+  for (auto it = isolators_.cbegin(), ed = isolators_.end(); it != ed; ++it) {
```
Comment (slava77) [inline PhysicsTools/PatAlgos/src/MultiIsolator.cc:92]: a for range loop could be used here

**Review by slava77 — COMMENTED** [review]
this will not block the reco signature:

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/30311	a48c030 
- technical
- jenkins tests pass

**Comment by silviodonato** [comment]
+1

**Pull request merged**